src/engine/tools.cpp

```diff
@@ -66,6 +66,13 @@ std::string StringLower( std::string str )
     return str;
 }
 
+/* convert to upper case */
+std::string StringUpper( std::string str )
```

Comment:
```suggestion
std::string StringUpper( const std::string & str )
```
Looking around, other similar functions could also use this signature.